sockeye/beam_search.py

```diff
@@ -72,11 +72,19 @@ def decode_step(self,
                     step_input: mx.nd.NDArray,
                     states: List,
                     vocab_slice_ids: Optional[mx.nd.NDArray] = None):
-        logits, states, _ = self._model.decode_step(step_input, states, vocab_slice_ids)
+        logits, states, target_factor_outputs = self._model.decode_step(step_input, states, vocab_slice_ids)
         if not self._skip_softmax:
             logits = logits.log_softmax(axis=-1)
         scores = -logits
-        return scores, states
+
+        target_factors = None  # type: Optional[mx.nd.NDArray]
```

Comment:
not super important, but given that this is identical to the ensemble case would it make sense to combine both into a single method (say in _Inference?)

Reply:
They are ALMOST the same. The ensemble case uses argmin and the interpolation method, whereas the single model case uses argmax. I'll leave it as is for now.